src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/WrapWithTag/WrapWithTagEndpointTests.cs

```diff
@@ -0,0 +1,202 @@
+                TextDocument = new TextDocumentIdentifier(uri)
+            };
+
+            // Act
+            var result = await endpoint.Handle(wrapWithDivParams, CancellationToken.None);
+
+            // Assert
+            Assert.NotNull(result);
+            languageServer.Verify();
+        }
+
+        [Fact]
+        public async Task Handle_CSharp_ReturnsNull()
+        {
+            // Arrange
+            var codeDocument = TestRazorCodeDocument.Create("@counter");
+            var uri = new Uri("file://path/test.razor");
+            var documentResolver = CreateDocumentResolver(uri.GetAbsoluteOrUNCPath(), codeDocument);
+            var responseRouterReturns = new Mock<IResponseRouterReturns>(MockBehavior.Strict);
+            responseRouterReturns
+                .Setup(l => l.Returning<WrapWithTagResponse>(It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new WrapWithTagResponse());
+
+            var languageServer = new Mock<ClientNotifierServiceBase>(MockBehavior.Strict);
+            languageServer
+                .Setup(l => l.SendRequestAsync(LanguageServerConstants.RazorWrapWithTagEndpoint, It.IsAny<WrapWithTagParams>()))
+                .ReturnsAsync(responseRouterReturns.Object);
+
+            var documentMappingService = Mock.Of<RazorDocumentMappingService>(
+                s => s.GetLanguageKind(codeDocument, It.IsAny<int>()) == RazorLanguageKind.CSharp, MockBehavior.Strict);
+            var endpoint = new WrapWithTagEndpoint(
+                languageServer.Object,
+                Dispatcher,
+                documentResolver,
+                documentMappingService,
+                LoggerFactory);
+
+            var wrapWithDivParams = new WrapWithTagParams()
+            {
+                Range = new(0, 0, 0, 2),
+                TextDocument = new TextDocumentIdentifier(uri)
+            };
+
+            // Act
+            var result = await endpoint.Handle(wrapWithDivParams, CancellationToken.None);
+
+            // Assert
+            Assert.Null(result);
+            languageServer.Verify();
+        }
+
+        [Fact]
+        public async Task Handle_DocumentNotFound_ReturnsNull()
+        {
+            // Arrange
+            var codeDocument = TestRazorCodeDocument.Create("<div></div>");
+            var uri = new Uri("file://path/test.razor");
+            var documentResolver = CreateDocumentResolver(uri.GetAbsoluteOrUNCPath(), codeDocument);
+            var responseRouterReturns = new Mock<IResponseRouterReturns>(MockBehavior.Strict);
+            responseRouterReturns
+                .Setup(l => l.Returning<WrapWithTagResponse>(It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new WrapWithTagResponse());
+
+            var languageServer = new Mock<ClientNotifierServiceBase>(MockBehavior.Strict);
+            languageServer
+                .Setup(l => l.SendRequestAsync(LanguageServerConstants.RazorWrapWithTagEndpoint, It.IsAny<WrapWithTagParams>()))
```

Comment:
If I understand correctly this shouldn't actually be called, shouldn't we therefore not setup the mock?

Reply:
Yeah, makes sense